flag/file_test.go

```diff
@@ -0,0 +1,94 @@
+package flag
+
+import (
+	"errors"
+	goflag "flag"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/urso/ucfg"
+)
+
+func TestFlagFileParsePrimitives(t *testing.T) {
+	config, err := parseTestFileFlags("-c test.a -c test.b",
+		map[string]FileLoader{
+			".a": makeLoadTestConfig(map[string]interface{}{
+				"b": true,
+				"i": 42,
+			}),
+			".b": makeLoadTestConfig(map[string]interface{}{
+				"b2": true,
+				"f":  3.14,
+				"s":  "string",
+			}),
+		},
+	)
+
+	assert.NoError(t, err)
+	checkFields(t, config)
+}
+
+func TestFlagFileParseOverwrites(t *testing.T) {
+	config, err := parseTestFileFlags("-c test.a -c test.b",
```

Comment:
Thanks for the detailed explanation. Would be great to have these also in the docs somewhere.

About future requests: We were talking about injecting logging in the past. As soon as we have that it would be nice to have a log message when values are overwritten.